x/update_validators/init_test.go

```diff
@@ -0,0 +1,63 @@
+package update_validators
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/confio/weave"
+	"github.com/confio/weave/store"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestInitState(t *testing.T) {
+	Convey("Test init state", t, func() {
+		// test data
+		addr := []byte("12345678901234567890")
+		addr2 := []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x30}
+		accts := WeaveAccounts{[]weave.Address{addr}}
+
+		accountsJson, err := json.Marshal(accts)
+		So(err, ShouldBeNil)
+
+		accountsJson2 := []byte(`{"addresses":["0102030405060708090021222324252627282930"]}`)
+		accts2 := WeaveAccounts{[]weave.Address{addr2}}
+
+		init := Initializer{}
+
+		kv := store.MemStore()
+		bucket := NewBucket()
+
+		Convey("Init works with no appState data", func() {
```

Comment:
Nice test cases. I like you covering all the main cases.
From first glance, convey doesn't seem bad, but doesn't convince me it is better than testify...

The main difference is running all test cases one after another vs. table tests... and i agree table tests can get quite ugly, but are easier to add

Reply:
Yes, but they need extra effort debugging.
Convey is good because if you look at it closely you will see that it runs the outer test body for each inner test, so basically it does the setUp automagically for each test. Problem is in that test case this is not helping much.

Reply:
So all that setup code is run once for each inner function?
I'll wait for a more complex example to really see the value.
Maybe in tools, where there is setup to start a server, make a connection to it, and then perform actions.